tf_agents/utils/nest_utils.py

```diff
@@ -554,5 +560,10 @@ def where(condition, true_outputs, false_outputs):
     Interleaved output from `true_outputs` and `false_outputs` based on
     `condition`.
   """
-  return tf.nest.map_structure(lambda t, f: tf.compat.v1.where(condition, t, f),
-                               true_outputs, false_outputs)
+  def _where_with_matching_ranks(t, f):
```

Comment:
thank you for refactoring this part of the code. can you lastly add a comment about what this helper function does?

Reply:
Hi @egonina , see the docstring I added to the helper. Thanks for all your feedback.